### PR TITLE
Pass through server implementation parameter to puppet class

### PR DESCRIPTION
**Pull request by stbenjam**
We need to expose this for the upgrade from puppet 3 to 4 (instructions in progress @ bottom of http://pad-katello.rhcloud.com/p/puppet4)

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -49,31 +52,33 @@
...
+  $certs_tar                    = $capsule::params::certs_tar,
+  $pulp_master                  = $capsule::params::pulp_master,
+  $pulp_admin_password          = $capsule::params::pulp_admin_password,
+  $pulp_oauth_effective_user    = $capsule::params::pulp_oauth_effective_user,
+  $pulp_oauth_key               = $capsule::params::pulp_oauth_key,
+  $pulp_oauth_secret            = $capsule::params::pulp_oauth_secret,
+
+  $puppet                       = $capsule::params::puppet,
+  $puppet_ca_proxy              = $capsule::params::puppet_ca_proxy,
+  $puppet_server_implementation = $capsule::params::puppet_server_implementation,
```
Comment (ekohl) [inline manifests/init.pp]: https://github.com/Katello/puppet-capsule/pull/90/files?w=1 for the lazy :)

File: manifests/params.pp [hunk 2]
```diff
@@ -19,6 +19,19 @@
...
 
+  if versioncmp($::puppetversion, '4.0') < 0 {
+    $aio_package      = false
+  } elsif $::rubysitedir =~ /\/opt\/puppetlabs\/puppet/ {
+    $aio_package      = true
+  } else {
+    $aio_package      = false
+  }
+
+  $server_implementation = $aio_package ? {
```
Comment (ekohl) [inline manifests/params.pp]: If we leave this as undef, will that work as well?

Comment (stbenjam) [inline manifests/params.pp]: If you pass an undef variable to a class, does it take the default instead?

Comment (ekohl) [inline manifests/params.pp]: It does. I think the installer handles this as well

**Comment by beav** [comment]
I was able to upgrade from puppet 3 to 4 after applying this patch and running through the upgrade procedure.

**Comment by ekohl** [comment]
ACK

**Comment by beav** [comment]
@ekohl thanks!

@stbenjam is away, merging on his behalf.

**Pull request merged**